### PR TITLE
Test Leapp RPM in CI

### LINKED ISSUE #245: Test the `leapp` RPM in CI
Currently the CI tests use `pipsi` and symlinks to run both the Cockpit plugin and `leapp-tool` itself directly from the git clone.

We want to change that so it tests the RPM-installed version in CI (while still supporting running the tests directly against the git clone locally), so this is my proposal for how to achieve that:

- [x] Add a `LEAPP_TEST_RPM` environment variable check to the `before_all` hook in the integration tests and note the result on the test context
- [x] Skip the `pipsi` invocation in the setup if that environment variable is set
- [x] Skip the creation of the user level Cockpit plugin symlinks if that environment variable is set
- [x] Update the integration testing Vagrant/Ansible configuration so it's possible to log in to those VMs as root using the default testing key (not just as vagrant)
- [x] Install the RPM after building it in `centos-ci/bootstrap.sh` (already implemented previously)
- [x] Run the remote-authentication feature tests without setting `LEAPP_TEST_RPM` in `centos-ci/bootstrap.sh` as a smoke test prior to running the full tests against the RPM
- [x] Set `LEAPP_TEST_RPM=1` when running the full tests in `centos-ci/bootstrap.sh`
- [x] Update the integration testing documentation accordingly

### PR DESCRIPTION
**Pull request by ncoghlan**
Closes #245

### CODE REVIEW

**Comment by ncoghlan** [comment]
Flagged as WIP as at least docs updates are still needed

**Comment by ncoghlan** [comment]
Removed the WIP marker as I've covered all the points identified in #245.

That doesn't mean there won't be further changes based on CI results and review comments, though :)

File: docs/source/integration-tests.rst [hunk 1]
```diff
@@ -1,8 +1,8 @@
 Integration tests
 =================
 
-Kernel modernization integration tests
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Legacy Application import integration tests
```
Comment (vinzenz) [inline docs/source/integration-tests.rst]: Thanks I noticed that already the other day and forgot to post a patch later

**Comment by ncoghlan** [comment]
Ah, just noticed that this is going to use the wrong `leapp-tool` for most of the testing, since the tests are currently still hardcoded to use the pipsi-installed version.

I need to move the calculation of the path to `leapp-tool` into the ClientHelper constructor.

File: integration-tests/features/steps/cockpit_demo.py [hunk 2]
```diff
@@ -18,16 +18,21 @@ def create_demonstration_user(context):
...
-    assert_that(os.path.exists(user_plugin_dir), "User plugin not installed")
-    # Check the rest of the repo is linked where the plugin will find it
-    leapp_dir_link = pathlib.Path("/opt/leapp")
-    desired_leapp_target = context.BASE_REPO_DIR
-    _ensure_expected_link(leapp_dir_link, desired_leapp_target)
+    """Ensures demo plugin is available to demo user"""
+    if context.TESTING_RPM:
+        # Plugin should be installed globally
+        assert_that(os.path.exists("/usr/bin/leapp-tool"), "LeApp CLI not installed")
+        assert_that(os.path.exists("/var/lib/cockpit/leapp"), "LeApp plugin not installed")
```
Comment (vinzenz) [inline integration-tests/features/steps/cockpit_demo.py]: /usr/share/cockpit/leapp

**Comment by ncoghlan** [comment]
The approach I went with to resolve the tool path issue was to update ClientHelper to accept it as a constructor parameter, and the pipsi install helper to return the destination path it installs to.

**Comment by ncoghlan** [comment]
While that passed, I'm not happy with the extra CI runtime introduced by the second behave call - I'm going to push an update that leaves the affected test VM running between the two invocations.

**Pull request merged**